code/game/objects/effects/particle_holder.dm

```diff
@@ -12,24 +12,29 @@
 	/// See \code\__DEFINES\particles.dm
 	var/particle_flags = NONE
 
-	var/atom/parent
+	var/datum/weakref/parent_ref
 
 /obj/effect/abstract/particle_holder/Initialize(mapload, particle_path = /particles/smoke, particle_flags = NONE)
 	. = ..()
 	if(!loc)
 		stack_trace("particle holder was created with no loc!")
 		return INITIALIZE_HINT_QDEL
 	// We nullspace ourselves because some objects use their contents (e.g. storage) and some items may drop everything in their contents on deconstruct.
-	parent = loc
+	parent_ref = WEAKREF(loc)
 	loc = null
 
 	// Mouse opacity can get set to opaque by some objects when placed into the object's contents (storage containers).
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
 	src.particle_flags = particle_flags
 	particles = new particle_path()
 	// /atom doesn't have vis_contents, /turf and /atom/movable do
-	var/atom/movable/lie_about_areas = parent
-	lie_about_areas.vis_contents += src
+	var/atom/movable/parent = parent_ref.resolve()
+	if(!parent)
```

Comment:
`isnull()` instead here if `resolve()` somehow returns something that is not null i would rather eat the runtimes

Reply:
isnull is preferable because it is
A) more explicit in the intent and reading of the code, which is always desirable
B) it's slightly, slightly, slightly, faster than just raw-calling `!`. this isn't the primary reason we use it but it's a great addition that really seals the deal on using it

if you're guaranteed to get a null value (like in weakref resolution), it's best to use `isnull()` for these sorts of things.